### PR TITLE
Splash screen

### LINKED ISSUE #7: [UI] Configure App Splash Screen
- Configure the app splash screen from the default expo splash screen to the one laid out on the [design](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/specs/) 
- Follow [contribution guidelines](https://github.com/droidconKE/droidconKE2022ReactNative/blob/develop/CONTRIBUTING.md) laid out in our contribution guide

**Additional context**
Screenshot 2022-09-12 at 20 27 14

### PR DESCRIPTION
**Pull request by mog-rn**
# Description

I have been able to add the splash screen for the app.  I have also updated the path for the assets on app.json since they were wrong.

Fixes #7 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

To test the changes I have been able to run the app on expo and the splash screen is displayed. I am not good with storybook so I have not been able to write any tests  for this.

Screenshot_20220923-043405

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mog-rn** [comment]
@brianwachira you can review this 🙏🏽

File: App.tsx [hunk 1]
```diff
@@ -5,6 +5,10 @@ import { NavigationContainer } from "@react-navigation/native";
 import useCachedResources from "./src/hooks/useCachedResources";
 import MainStackNavigator from "./src/navigation/MainStackNavigator";
 
+import * as SplashScreen from "expo-splash-screen";
+
+SplashScreen.preventAutoHideAsync();
+
```
Comment (brianwachira) [inline App.tsx:11]: Good work!
This may not be necessary as Expo will automatically load the splash screen when needed. 
You can remove it.

**Review by brianwachira — CHANGES_REQUESTED** [review]
Good work so far!
I have noticed the src/assets/splash.png does not match the one in the design [https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/specs/](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/specs/)

I think the Design team changed it last minute. Could you update the image to use the one on design?

I also noticed the one on design doesn't have the same dimensions as the one provided by Expo so I created a separate 1 with the same dimensions as the one provided by Expo. You can check it out here [https://www.figma.com/file/iCiXKs2kmXc6fqtMJb5l46/App-Icon-%26-Splash-(Copy)?node-id=80208%3A50](https://www.figma.com/file/iCiXKs2kmXc6fqtMJb5l46/App-Icon-%26-Splash-(Copy)?node-id=80208%3A50)

File: App.tsx (hunk 1 above)
Comment (brianwachira) [inline App.tsx]: Good work!
This may not be necessary as Expo will automatically load the splash screen when needed. 
You can remove it.

**Review by brianwachira — CHANGES_REQUESTED** [review]

File: app.json [hunk 2]
```diff
@@ -22,12 +22,12 @@
...
     "android": {
       "adaptiveIcon": {
-        "foregroundImage": "./assets/adaptive-icon.png",
+        "foregroundImage": "./src/assets/adaptive-icon.png",
         "backgroundColor": "#FFFFFF"
       }
     },
     "web": {
-      "favicon": "./assets/favicon.png"
+      "favicon": "../assets/favicon.png"
```
Comment (makunomark) [inline app.json]: **Question**: Should this be `./src/assets/favicon.png`?

```suggestion
      "favicon": "./src/assets/favicon.png"
```

File: App.tsx [hunk 3]
```diff
@@ -5,6 +5,10 @@ import { NavigationContainer } from "@react-navigation/native";
 import useCachedResources from "./src/hooks/useCachedResources";
 import MainStackNavigator from "./src/navigation/MainStackNavigator";
 
+import * as SplashScreen from "expo-splash-screen";
+
+SplashScreen.preventAutoHideAsync();
```
Comment (makunomark) [inline App.tsx]: **Question**: According to the [docs](https://docs.expo.dev/versions/latest/sdk/splash-screen/#splashscreenpreventautohideasync), it seems like `preventAutoHideAsync` will prevent react native from closing the splash screen until [`hideAsync()`](https://docs.expo.dev/versions/latest/sdk/splash-screen/#splashscreenpreventautohideasync)) is called. Is this something we should add somewhere along the lifecycle of App.tsx?

File: app.json (hunk 2 above)
Comment (mog-rn) [reply app.json]: yes it should @makunomark

**Comment by mog-rn** [comment]
@brianwachira I have been able to resolve the issues that you had raised

File: App.tsx (hunk 3 above)
Comment (mog-rn) [reply App.tsx]: I believe this was handled in the useCachedResources() hook so I think it is not that necessary to do it again @makunomark

**Review by mog-rn — COMMENTED** [review]
I have been able to resolve all the changes that were requested @makunomark @brianwachira

File: App.tsx (hunk 3 above)
Comment (makunomark) [reply App.tsx]: Ah okay. Was curious.

Comment (makunomark) [reply App.tsx]: Possibly not an issue with this Pr, however, I have seen where we prevent the screen from closing, in useCachedResources. From the same docs, it seems like the code that prevents the SplashScreen from closing should not be placed inside a hook or a component. 

(Forgive me for not posting the links, I'm on the move on my phone)

It's a conversation we can pick up on an issue or something. Shouldn't prevent the merging of this PR.

**Review by brianwachira — APPROVED** [review]

**Review by brianwachira — APPROVED** [review]

**Review by brianwachira — APPROVED** [review]

**Pull request merged**